moped-editor/src/layouts/DashboardLayout/NavBar/NavigationSearchInput.js

```diff
@@ -0,0 +1,246 @@
+    padding: "2px",
+    [theme.breakpoints.up("sm")]: {
+      width: "300px",
+    },
+    [theme.breakpoints.down("sm")]: {
+      width: "200px",
+    },
+    height: "36px",
+  },
+  inputInput: {
+    [theme.breakpoints.up("sm")]: {
+      fontSize: "0.875rem",
+    },
+    [theme.breakpoints.down("sm")]: {
+      fontSize: "0.75rem",
+    },
+    paddingLeft: "1em",
+  },
+  adornedStart: {
+    color: theme.palette.text.secondary,
+    paddingLeft: "5px",
+  },
+  inputFocused: {
+    borderWidth: "2px",
+    borderRadius: "4px",
+    borderStyle: "solid",
+    borderColor: theme.palette.primary.main,
+  },
+  searchPopper: {
+    borderWidth: "1px",
+    borderRadius: "4px",
+    borderStyle: "solid",
+    borderColor: theme.palette.background.default,
+    [theme.breakpoints.up("sm")]: {
+      width: "300px",
+    },
+    [theme.breakpoints.down("sm")]: {
+      width: "200px",
+    },
+    boxShadow: "0 0 1px 0 rgb(0 0 0 / 31%), 0 3px 3px -3px rgb(0 0 0 / 25%)",
+  },
+  searchResults: {
+    backgroundColor: theme.palette.background.paper,
+    borderRadius: "4px",
+    color: theme.palette.text.primary,
+  },
+  tempResults: {
+    padding: "16px",
+  },
+}));
+
+/**
+ * @return {JSX.Element}
+ * @constructor
+ */
+const NavigationSearchInput = () => {
+  const classes = useStyles();
+  const divRef = React.useRef();
+  let projectSearchQuery = new GQLAbstract(ProjectsListViewQueryConf);
+
+  // Toggle Text Input or magnifying glass
+  const [searchInput, showSearchInput] = useState(false);
+  // anchor element for results popper to "attach" to
+  const [searchResultsAnchor, setSearchResultsAnchor] = useState(null);
+  const [searchTerm, setSearchTerm] = useState("");
+
+  const [loadSearchResults, { called, loading, data }] = useLazyQuery(
+    projectSearchQuery.gql,
+    projectSearchQuery.config.options.useQuery
+  );
+
+  const handleMagClick = () => {
+    showSearchInput(true);
+  };
+
+  const handleDropdownClose = () => {
+    setSearchResultsAnchor(null);
+    showSearchInput(false);
+    setSearchTerm("");
+  };
+
+  const handleSearchFocus = () => {
+    setSearchResultsAnchor(divRef.current);
+  };
+
+  /**
+   * Handles special keys typed in the search bar
+   * @param {string} key - The key name being typed
+   */
+  const handleKeyDown = key => {
+    switch (key) {
+      // On Escape key, clear the search
+      case "Escape":
+        setSearchTerm("");
+        setSearchResultsAnchor(null);
+        break;
+      // On Enter key, initialize the search
+      case "Enter":
+        handleSearchSubmission(null);
+        break;
+
+      default:
+        return;
+    }
+  };
+
+  /**
+   * Formats search term and triggers lazy query to load search results
+   * @param {event} - Event that triggered submission
+   */
+  const handleSearchSubmission = event => {
+    // Stop if we don't have any value entered in the search field
+    if (searchTerm.length === 0) {
+      console.log("no search term");
+    }
```

Comment:
Minor thing, probably best to remove the console.log and replace with a `return;` statement? It's obvious that it's harmless but it would seem to me that the return statement would at least prevent the rest of the code from being executed if there is nothing to search for.